src/crypto/OlmDevice.js

```diff
@@ -115,12 +115,29 @@ export function OlmDevice(cryptoStore) {
  * found.
  *
  * Reads the device keys from the OlmAccount object.
+ *
+ * @param {object} fromExportedDevice (Optional) data from exported device
+ *     that must be re-created.
  */
-OlmDevice.prototype.init = async function() {
+OlmDevice.prototype.init = async function(fromExportedDevice) {
```

Comment:
Since the device is exported using the pickle key, there should probably be a way to set the pickle key (when not imported from an exported key).  Maybe it makes sense to make the argument an `opts` object argument that you can use to set a pickle key or pass the exported device.  (And then pass it in from the `MatrixClient` initialization.)

Reply:
So you would suggest to start this method with a `const { pickleKey, fromExportedDevice } = opts`?

Right now `fromExportedDevice` contains the pickleKey, so I would see two different strategy:
1. the caller of the method has to split the exported data before passing them to the method (and `fromExportedDevice` does **not** contain the pickle key any more
2. the caller still passes `fromExportedDevice` unmodified and with a pickleKey, in which case this method will throw in case the pickle key is passed both through `fromExportedDevice` and as another keyword argument.

Are we on the same page and which strategy would you prefer?

Reply:
Yes.  Strategy 2 was what I was thinking of (or perhaps ignore `opts.pickleKey` and print a warning, instead of doing a `throw`)